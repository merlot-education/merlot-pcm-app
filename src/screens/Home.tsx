import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import { useTranslation } from 'react-i18next'
import { Colors, TextTheme } from '../theme/theme'
import Button, { ButtonType } from '../components/button/Button'
import Screens from '../utils/constants'

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.background,
    margin: 20,
  },
  bodyText: {
    ...TextTheme.normal,
    flexShrink: 1,
  },
  spacer: {
    height: 40,
  },
})

interface HomeProps {
  navigation: any
}

const Home: React.FC<HomeProps> = ({ navigation }) => {
  const { t } = useTranslation()

  const nextPage = () => {
    navigation.navigate(Screens.Scan)
  }

  return (
    <View style={[styles.container]}>
      <Text style={[styles.bodyText, { fontWeight: 'bold' }]}>
        {/* {t('Home.LoginMsg')} */}
        Your wallet is created successfully.
      </Text>
      <View style={styles.spacer} />
      <Button
        title={t('Global.Continue')}
        buttonType={ButtonType.Primary}
        onPress={nextPage}
      />
    </View>
  )
}

export default Home
