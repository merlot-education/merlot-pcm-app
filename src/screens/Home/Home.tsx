import React from 'react';
import {
  Dimensions,
  FlatList,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { useTranslation } from 'react-i18next';
import { StackScreenProps } from '@react-navigation/stack';
import { CredentialState } from '@aries-framework/core';
import { useCredentialByState } from '@aries-framework/react-hooks';
import { ColorPallet, TextTheme } from '../../theme/theme';
import useNotifications from '../../hooks/notifications';
import { HomeStackParams, Screens } from '../../types/navigators';
import InfoTextBox from '../../components/text/InfoTextBox';
import { NotificationListItem } from '../../components';
import { NotificationType } from '../../components/listItems/NotificationListItem';

const { width } = Dimensions.get('window');

type HomeProps = StackScreenProps<HomeStackParams, Screens.Home>;

const Home: React.FC<HomeProps> = ({ navigation }) => {
  const { t } = useTranslation();
  const { notifications } = useNotifications();
  const credentials = [
    ...useCredentialByState(CredentialState.CredentialReceived),
    ...useCredentialByState(CredentialState.Done),
  ];
  const emptyListComponent = () => (
    <View style={{ marginHorizontal: offset, width: width - 2 * offset }}>
      <InfoTextBox showIcon>
        <Text style={TextTheme.normal}>{t('Home.NoNewUpdates')}</Text>
      </InfoTextBox>
    </View>
  );

  const displayMessage = (credentialCount: number) => {
    if (typeof credentialCount === 'undefined') {
      throw new Error(t('Home.CredentialCountUndefinedError'));
    }

    let credentialMsg;

    if (credentialCount === 1) {
      credentialMsg = (
        <Text>
          {t('Home.YouHave')}{' '}
          <Text style={{ fontWeight: 'bold' }}>{credentialCount}</Text>{' '}
          {t('Home.Credential')} {t('Home.InYourWallet')}
        </Text>
      );
    } else if (credentialCount > 1) {
      credentialMsg = (
        <Text>
          {t('Home.YouHave')}{' '}
          <Text style={{ fontWeight: 'bold' }}>{credentialCount}</Text>{' '}
          {t('Home.Credentials')} {t('Home.InYourWallet')}
        </Text>
      );
    } else {
      credentialMsg = t('Home.NoCredentials');
    }

    return (
      <View style={[styles.messageContainer]}>
        {credentialCount === 0 ? (
          <Text
            style={[
              TextTheme.headingTwo,
              { marginTop: offset, marginBottom: 35 },
            ]}>
            {t('Home.Welcome')}
          </Text>
        ) : null}
        <Text
          style={[
            TextTheme.normal,
            { marginTop: offset, textAlign: 'center' },
          ]}>
          {credentialMsg}
        </Text>
      </View>
    );
  };

  return (
    <View>
      <View style={styles.rowContainer}>
        <Text style={[TextTheme.headingFour, styles.header]}>
          {t('Home.Notifications')}
          {notifications.length ? ` (${notifications.length})` : ''}
        </Text>
        {notifications?.length > 1 ? (
          <TouchableOpacity
            style={styles.linkContainer}
            activeOpacity={1}
            onPress={() => navigation.navigate(Screens.Notifications)}>
            <Text style={styles.link}>{t('Home.SeeAll')}</Text>
          </TouchableOpacity>
        ) : null}
      </View>
      <FlatList
        horizontal
        showsHorizontalScrollIndicator={false}
        scrollEnabled={notifications.length > 0}
        snapToOffsets={[
          0,
          ...Array(notifications.length)
            .fill(0)
            .map(
              (n: number, i: number) =>
                i * (width - 2 * (offset - offsetPadding)),
            )
            .slice(1),
        ]}
        decelerationRate="fast"
        ListEmptyComponent={emptyListComponent()}
        data={notifications}
        keyExtractor={item => item.id}
        renderItem={({ item, index }) => (
          <View
            style={{
              width: width - 3.5 * offset,
              marginLeft: !index ? offset : offsetPadding,
              marginRight:
                index === notifications.length - 1 ? offset : offsetPadding,
            }}>
            {item.type === 'CredentialRecord' ? (
              <NotificationListItem
                key={item.id}
                notificationType={NotificationType.CredentialOffer}
                notification={item}
              />
            ) : (
              <NotificationListItem
                key={item.id}
                notificationType={NotificationType.ProofRequest}
                notification={item}
              />
            )}
          </View>
        )}
      />
      <View style={styles.container}>{displayMessage(credentials.length)}</View>
    </View>
  );
};

export default Home;

const offset = 25;
const offsetPadding = 5;

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: offset,
  },
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: offset,
  },
  messageContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 35,
    marginHorizontal: 60,
  },
  header: {
    marginTop: offset,
    marginBottom: 20,
  },
  linkContainer: {
    minHeight: TextTheme.normal.fontSize,
    marginTop: 10,
  },
  link: {
    ...TextTheme.normal,
    color: ColorPallet.brand.link,
  },
});
