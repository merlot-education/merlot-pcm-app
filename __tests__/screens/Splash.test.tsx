import React from 'react'
import { create } from 'react-test-renderer'
import Home from '../../src/screens/Splash'

jest.mock('react-native-splash-screen', () => 'Splash')
jest.mock('react-native-keychain', () => 'Splash')

describe('Splash Screen', () => {
  it('Renders correctly', () => {
    const tree = create(<Home />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
