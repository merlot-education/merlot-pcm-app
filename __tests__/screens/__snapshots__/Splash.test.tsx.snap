// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash Screen Renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
/>
`;
